### PR TITLE
items: Adapts JSON schema for temporary item type

### PR DESCRIPTION
**Pull request by zannkukai**
Adapts the item JSON schema and ES mapping to introduce the 'temporary
item type` entry.

Co-Authored-by: Renaud Michotte <renaud.michotte@gmail.com>

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?
- [ ] Cypress tests successful?

### CODE REVIEW

**Review by iGormilhit — APPROVED** [review]

File: rero_ils/modules/items/api/record.py [hunk 1]
```diff
@@ -72,6 +77,18 @@ def extended_validation(self, **kwargs):
         if len(note_types) != len(set(note_types)):
             return _('Can not have multiple notes of same type.')
 
+        # check temporary item type data
+        tmp_itty = self.get('temporary_item_type')
+        if tmp_itty:
```
Comment (BadrAly) [inline rero_ils/modules/items/api/record.py:82]: Does this include also the possibility that a librarian changes the `item_type` to equal the `temporary_item_type`?

File: rero_ils/modules/items/jsonschemas/items/item-v0.0.1.json [hunk 2]
```diff
@@ -123,16 +160,73 @@
...
+              "hideLabel": true
+            },
+            "remoteOptions": {
+              "type": "item_types"
+            }
+          }
+        },
+        "end_date": {
+          "title": "End date",
+          "description": "If set, specifies the date when the temporary circulation category will be disabled.",
```
Comment (BadrAly) [inline rero_ils/modules/items/jsonschemas/items/item-v0.0.1.json:206]: `will be removed`?

File: rero_ils/modules/items/mappings/v7/items/item-v0.0.1.json [hunk 3]
```diff
@@ -91,6 +91,19 @@
           }
         }
       },
+      "temporary_item_type": {
+        "properties": {
+          "type": {
```
Comment (BadrAly) [inline rero_ils/modules/items/mappings/v7/items/item-v0.0.1.json:96]: Do you need this field `type` here?

File: tests/unit/test_items_jsonschema.py [hunk 4]
```diff
@@ -98,3 +100,26 @@ def test_new_acquisition(item_schema, item_lib_martigny_data_tmp):
         acq_date = datetime.date.today().strftime('%Y/%m/%d')
         item_lib_martigny_data_tmp['acquisition_date'] = acq_date
         validate(item_lib_martigny_data_tmp, item_schema)
+
+
+def test_temporary_item_type(item_schema, item_lib_martigny):
+    """Test temporary item type for item jsonschemas."""
+    data = item_lib_martigny
+
+    # tmp_itty cannot be the same than main_itty
```
Comment (BadrAly) [inline tests/unit/test_items_jsonschema.py:109]: Need the other test, when librarian changes `item_type` to equal the `temp_item_type`

**Review by BadrAly — CHANGES_REQUESTED** [review]

File: rero_ils/modules/items/mappings/v7/items/item-v0.0.1.json (hunk 3 above)
Comment (zannkukai) [reply rero_ils/modules/items/mappings/v7/items/item-v0.0.1.json:96]: all '$ref' has this structure into all ES mapping ;-)

File: rero_ils/modules/items/jsonschemas/items/item-v0.0.1.json (hunk 2 above)
Comment (zannkukai) [reply rero_ils/modules/items/jsonschemas/items/item-v0.0.1.json:206]: This description seems important ? @iGormilhit helps me to get a correct sentence. @iGormilhit, can you confirm my feeling ?

File: rero_ils/modules/items/api/record.py (hunk 1 above)
Comment (zannkukai) [reply rero_ils/modules/items/api/record.py:82]: Yes it's cover. If librarian change default `itty` to use the same value than `tmp_itty` this validation rules will raised an error. In this case, the solution for librarian is to remove the `tmp_itty` value(s)

File: tests/unit/test_items_jsonschema.py (hunk 4 above)
Comment (zannkukai) [reply tests/unit/test_items_jsonschema.py:109]: It will raised the same error. No need write another test.

**Review by BadrAly — APPROVED** [review]

**Review by rerowep — APPROVED** [review]

File: rero_ils/modules/items/jsonschemas/items/item-v0.0.1.json (hunk 2 above)
Comment (iGormilhit) [reply rero_ils/modules/items/jsonschemas/items/item-v0.0.1.json:206]: [Un]fortunately, @BadrAly is right. Sorry. :slightly_smiling_face:

**Pull request merged**